mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp

```diff
@@ -1546,24 +1546,32 @@ class VectorPrintOpConversion : public ConvertOpToLLVMPattern<vector::PrintOp> {
 
     auto punct = printOp.getPunctuation();
     if (auto stringLiteral = printOp.getStringLiteral()) {
-      LLVM::createPrintStrCall(rewriter, loc, parent, "vector_print_str",
-                               *stringLiteral, *getTypeConverter(),
-                               /*addNewline=*/false);
+      if (LLVM::createPrintStrCall(rewriter, loc, parent, "vector_print_str",
+                                   *stringLiteral, *getTypeConverter(),
+                                   /*addNewline=*/false)
+              .failed()) {
+        return failure();
+      }
     } else if (punct != PrintPunctuation::NoPunctuation) {
-      emitCall(rewriter, printOp->getLoc(), [&] {
-        switch (punct) {
-        case PrintPunctuation::Close:
-          return LLVM::lookupOrCreatePrintCloseFn(parent);
-        case PrintPunctuation::Open:
-          return LLVM::lookupOrCreatePrintOpenFn(parent);
-        case PrintPunctuation::Comma:
-          return LLVM::lookupOrCreatePrintCommaFn(parent);
-        case PrintPunctuation::NewLine:
-          return LLVM::lookupOrCreatePrintNewlineFn(parent);
-        default:
-          llvm_unreachable("unexpected punctuation");
-        }
-      }());
+      if (auto op = [&] -> FailureOr<LLVM::LLVMFuncOp> {
+            switch (punct) {
+            case PrintPunctuation::Close:
+              return LLVM::lookupOrCreatePrintCloseFn(parent);
+            case PrintPunctuation::Open:
+              return LLVM::lookupOrCreatePrintOpenFn(parent);
+            case PrintPunctuation::Comma:
+              return LLVM::lookupOrCreatePrintCommaFn(parent);
+            case PrintPunctuation::NewLine:
+              return LLVM::lookupOrCreatePrintNewlineFn(parent);
+            default:
+              llvm_unreachable("unexpected punctuation");
+            }
+          }();
+          succeeded(op))
```

Comment:
Nit: As before, I would not wrap this into the if's condition. This is somewhat hard to read.